src/beignet/subsets/_murcko_scaffold_split.py

```diff
@@ -0,0 +1,122 @@
+import math
+import random
+from collections import defaultdict
+from typing import Sequence
+
+from torch.utils.data import Dataset, Subset
+
+try:
+    from rdkit import Chem
+    from rdkit.Chem.Scaffolds.MurckoScaffold import GetScaffoldForMol
+
+    _RDKit_AVAILABLE = True
+except (ImportError, ModuleNotFoundError):
+    _RDKit_AVAILABLE = False
+    Chem, MurckoScaffoldSmiles = None, None
+
+
+def murcko_scaffold_split(
+    dataset: Dataset,
+    smiles: Sequence[str],
+    test_size: float | int,
+    *,
+    seed: int = 0xDEADBEEF,
+    shuffle: bool = True,
+    include_chirality: bool = False,
+) -> tuple[Subset, Subset]:
```

Comment:
I think a better design is `List[Subset]` based on a `lengths` parameter, e.g., see `[torch.utils.data.random_split](https://pytorch.org/docs/stable/data.html#torch.utils.data.random_split)`.